src/gt4py/cartesian/frontend/exceptions.py

```diff
@@ -6,76 +6,88 @@
 # Please, refer to the LICENSE file in the root directory.
 # SPDX-License-Identifier: BSD-3-Clause
 
+from typing import Any
+
 from gt4py.cartesian import definitions as gt_definitions
+from gt4py.cartesian.frontend import nodes
 
 from . import gtscript_frontend
 
 
 class GTScriptSyntaxError(gt_definitions.GTSyntaxError):
-    def __init__(self, message, *, loc=None):
+    def __init__(self, message: str | None, *, loc: nodes.Location | None = None):
+        if message is None:
```

Comment:
Optional suggestion in case you also want to generate a proper message if an empty string is passed (it applies to all Exception classes in this file:
```suggestion
        if not message:
```
Additionally, you could implement the logic to add the location info in the base class, as a header or ending of the message passed from the subclasses. It's more work and it probably needs tweaking a bit the error messages so the final string makes sense, so it's up to you to decide, it's just an idea.